### ISSUE TITLE
CustomGap doesn't affect the final size of presented view

### ISSUE THREAD

**Issue by popei69** [Babylonpartners/DrawerKit#101]
I'm trying to place UI component depending of the size of the presented view, for instance having a button at the bottom of the drawer.

When testing it, it seems that the size of the presented view is never matching the size left from the custom gap. For instance, if I set a `.leavesCustomGap(gap: 100)`, the view presented frame won't get shorter of 100. Therefore any elements at the bottom might not be visible.

I guess what I would expect is that if the drawer can have a maximum size of 500, its presented view and contained one would reflect the same size. That's not the case at the moment. Is it something available with DrawerKit?

To replicate it, inspect the view frame in a presented view controller (in `viewDidAppear` for instance), it will be the size of the screen regardless the maximum size available of the drawer.

**Comment by ilyapuchka** [comment]
Indeed the position of the drawer in the fully expanded state is fixed and it does not consider the content. I have a similar issue, where I have to use partially expanded state to accommodate the content but there is no way then to disable fully ex expanded state

**Comment by ilyapuchka** [comment]
@popei69 is your issue that the drawer is not respecting the content size of the presented view or the other way around, that the content view is not respecting the size of the drawer?
To avoid having not visible elements you can try to put them in a scroll view. DrawerKit has support for that (see demo app) so that it works fine in terms of scrolling behaviour.

**Comment by popei69** [comment]
@ilyapuchka thanks for the feedback. The idea is actually to have a sticky button at all time at the bottom of the drawer regardless how far it is open, or at least when _half opened_ and _fully opened_. But the rest of the page would still be scrollable. To give you an idea, I'm looking for a similar behaviour as the Instagram Share Post feature. 
I just can't see a way to do that with AutoLayout and DrawerKit at the moment with a content size always describing the whole screen.

**Comment by ilyapuchka** [comment]
yep, I think out of the box it's not supported. You could try to hack in with animation behaviour, but I doubt it will really help.
I doubt it will work nicely with autolayout but you could make update the frame of the button manually somehow.
Speaking of autolayout it would be handy if drawer kit would provide some kind of layout guides that one could use to attache views to, or if it would adjust safe are of views involved in the presentation. For instance I had an issue when the button should be displayed in the repenting (background) view right above the drawer, it would maybe make it simpler if safe area of presenting view would be adjusted by the drawer, so that when it appears the button slides to the top with it. It might be tricky to make it universal though, but just adding layout guides in the drawer would help in your issue.

cc @inamiy @Ben-Henshall
